### PR TITLE
fix: add exception for decreasing_or_equal_stop_time_distance

### LINKED ISSUE #1882: Flex: create exception for decreasing_or_equal_stop_time_distance
### Describe the problem

[Based on this old feed](https://data.trilliumtransit.com/gtfs/mountadams-wa-us/mountadams-wa-us.zip) and [its associated validation report,](https://staging-gtfs-validator-results.mobilitydata.org/6a9c405f-11ab-4824-9b3d-6d77fa39d13f/report.html), there are deviated routes for this agency, where drivers can deviate off a fixed route to a specific zone based on demand booking. 

In this feed, there are cases where [`decreasing_or_equal_stop_time_distance`](https://gtfs-validator.mobilitydata.org/rules.html#decreasing_or_equal_stop_time_distance-rule) is triggered on stop times with `location_id` or `location_group_id` rather than `stop_id`.

This is confusing, because the `stop_id` column will be null in this case. 

Image

### Proposed solution

Create an exception for rows where a `shape_dist_traveled` value is provided and has either an associated `location_id` or `location_group_id`. These rows should be omitted and will be covered in #1885 

Sample datasets: 

https://data.trilliumtransit.com/gtfs/mountadams-wa-us/mountadams-wa-us.zip
http://data.trilliumtransit.com/gtfs/thurston-wa-us/thurston-wa-us--flex-v2.zip

### Alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**

The change ensures that the `DecreasingOrEqualStopTimeDistanceNotice` is not triggered in specific cases where certain identifiers are missing or present.

Closes #1882 
**Expected behavior:** 

Explain and/or show screenshots for how you expect the pull request to work in your testing (in case other devices exhibit different behavior).

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by emmambd** [comment]
LGTM on a QA side! Checked with https://data.trilliumtransit.com/gtfs/islandtransit-wa-us/islandtransit-wa-us--flex-v2-TEST.zip , http://data.trilliumtransit.com/gtfs/mountadams-wa-us/mountadams-wa-us--flex-v2.zip and http://data.trilliumtransit.com/gtfs/thurston-wa-us/thurston-wa-us--flex-v2.zip

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java [hunk 1]
```diff
@@ -44,14 +44,29 @@ public class StopTimeIncreasingDistanceValidator extends FileValidator {
...
   public void validate(NoticeContainer noticeContainer) {
     for (List<GtfsStopTime> stopTimeList : Multimaps.asMap(stopTimeTable.byTripIdMap()).values()) {
       // GtfsStopTime objects are sorted based on @SequenceKey annotation on stop_sequence field.
-      for (int i = 1; i < stopTimeList.size(); ++i) {
-        GtfsStopTime prev = stopTimeList.get(i - 1);
-        GtfsStopTime curr = stopTimeList.get(i);
-        if (prev.hasShapeDistTraveled()
+      GtfsStopTime prev = null;
+      GtfsStopTime curr;
+      for (int i = 0; i < stopTimeList.size(); ++i) {
```
Comment (jcpitre) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:59]: Nitpick: Now that it starts at 0 and i is not needed in the loop, `for (var curr : stopTimeList) {` would be enough.
Don't feel you have to change it.

Comment (davidgamez) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:59]: It's needed to set the `first` valid `prev` value needs to be set.

**Review by jcpitre — APPROVED** [review]
LGTM!

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java (hunk 1 above)
Comment (davidgamez) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:59]: Oh, I misread the feedback. Nice catch; yes, what you are proposing looks cleaner.

**Pull request merged**